launcher_abler/AblerLauncher.py

```diff
@@ -258,272 +258,66 @@ def __init__(self, parent=None):
         self.progressBar.setValue(0)
         self.progressBar.hide()
         self.lbl_task.hide()
-        self.statusbar.showMessage(f"Ready - Last check: {lastcheck}")
+        self.statusbar.showMessage(f"Ready - Last check: {self.lastcheck}")
         self.btn_Quit.clicked.connect(QtCore.QCoreApplication.instance().quit)
         self.btn_about.clicked.connect(self.about)
         self.btn_acon.clicked.connect(self.open_acon3d)
-        try:
-            if not (self.check_launcher()):
-                self.check_once()
-        except Exception as e:
-            logger.error(e)
 
-    def open_acon3d(self):
-        url = QtCore.QUrl("https://www.acon3d.com/")
-        QtGui.QDesktopServices.openUrl(url)
+    def setup_update_launcher_ui(self, finallist):
+        self.btn_update_launcher.show()
+        self.btn_update.hide()
+        self.btn_execute.hide()
+        self.btn_update_launcher.clicked.connect(
+            lambda throwaway=0, entry=finallist[0]: self.download(
+                entry, dir_name=launcherdir_
+            )
+        )
 
-    def about(self):
-        aboutText = (
-            '<html><head/><body><p>Utility to update ABLER to the latest version available at<br> \
-        <a href="https://builder.blender.org/download/"><span style=" text-decoration: underline; color:#2980b9;">\
-        https://builder.blender.org/download/</span></a></p><p><br/>Developed by Tobias Kummer for \
-        <a href="http://www.overmind-studios.de"><span style="text-decoration:underline; color:#2980b9;"> \
-        Overmind Studios</span></a></p><p>\
-        Licensed under the <a href="https://www.gnu.org/licenses/gpl-3.0-standalone.html"><span style=" text-decoration:\
-        underline; color:#2980b9;">GPL v3 license</span></a></p><p>Project home: \
-        <a href="https://overmindstudios.github.io/BlenderUpdater/"><span style=" text-decoration:\
-        underline; color:#2980b9;">https://overmindstudios.github.io/BlenderUpdater/</a></p> \
-        <p style="text-align: center;"></p> \
-        <p>Application based on the version: '
-            + launcher_installed
-            + "</p></body></html> "
+    def setup_update_abler_ui(self, finallist):
+        # ABLER를 업데이트
+        # TODO: 버튼 한번 클릭되면 비활성화 기능 넣기
```

Comment:
self.download에 self.btn_update enabled 해제시키는 기능을 추가하면 되지 않을까용

Reply:
넵 self.download에 내용 추가하면 될거 같습니다!

Reply:
```suggestion
    def setup_download_ui(self, entry, dir_name):
        ...
        # 중복 다운로드 방지를 위한 버튼 비활성화
        if dir_name == dir_:
            self.btn_update.setDisabled(True)
        else:
            self.btn_update_launcher.setDisabled(True)
```

해당 부분을 추가해서 업데이트 버튼 클릭하면 다운로드가 실행되면서 버튼이 비활성화 됩니다.
`config.ini` 업데이트 관련 내용은 조금 더 테스트해보고 새로운 PR로 진행할까 합니다.